### PR TITLE
store/tikv: remove canceled requests before sending

### PR DESCRIPTION
**Pull request by coocood**
### What problem does this PR solve? 
When requests are canceled, we still send the requests to TiKV.

### What is changed and how it works?
Remove the canceled requests before sending.

### Check List 

Tests 

 - Unit test

### CODE REVIEW

File: store/tikv/client.go [hunk 1]
```diff
@@ -506,6 +514,23 @@ func (a *connArray) batchSendLoop(cfg config.TiKVClient) {
 	}
 }
 
+// removeCanceledRequests removes canceled requests before sending.
+func removeCanceledRequests(
```
Comment (hicqu) [inline store/tikv/client.go]: How about write the logic in `fetchPendingRequest` to reduce allocaction?

Comment (coocood) [reply store/tikv/client.go]: It's possible that after wait for tikv overload, some of the requests canceled.
This way removes more canceled requests.

**Comment by coocood** [comment]
@lysu PTAL

**Comment by hicqu** [comment]
LGTM

**Comment by coocood** [comment]
@tiancaiamao PTAL

**Comment by tiancaiamao** [comment]
LGTM

**Pull request merged**